### ISSUE TITLE
humility qspi -{D, W} ... fails with StackOverflow in A0 on Gimlet-B

### ISSUE THREAD

**Issue by wesolows** [oxidecomputer/hubris#636]
If you happen to attempt to write the boot flash on a Gimlet-B while in A0, you get:

```
wesolows at monolith in ~
$ humility hiffy -c Sequencer.set_state -a state=A0
humility: attached via ST-Link V3
Sequencer.set_state() = ()
wesolows at monolith in ~
$ humility qspi -D ~/src/amd-host-image-builder/milan-gimlet-b.img
humility: attached via ST-Link V3
humility: erasing 16777216 bytes...
humility qspi failed: request failed: Some(Fault(StackOverflow))
```

Let's be clear: This should not work!  But the failure mode is surprising and seems likely to be accidental.  A better failure mode from hiffy seems desirable here.

**Comment by jclulow** [comment]
I am ostensibly hitting this in A2 right now (on `cadbury`):

```
$ pfexec humility -t sn9 hiffy -c Sequencer.get_state
humility: attached to 0483:374e:002F001A4741500520383733 via ST-Link V3
Sequencer.get_state() => A2

$ pfexec humility -t sn9 qspi -D rom
humility: attached to 0483:374e:002F001A4741500520383733 via ST-Link V3
humility: erasing 16252928 bytes...
humility qspi failed: request failed: Some(Fault(StackOverflow))
```

**Comment by cbiffle** [comment]
Did this get fixed? I feel like it must've gotten fixed or we'd be hearing about it. @bcantrill ?

**Comment by wesolows** [comment]
Pretty sure this was addressed by 0ab13dc7e86e3a4b3ab134f5d9964d637d8000be as #730 .  Haven't seen it in forever.
